bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml

```diff
@@ -206,7 +206,7 @@ spec:
                 - --enable-leader-election
                 command:
                 - /manager
-                image: quay.io/opentelemetry/opentelemetry-operator:v0.28.0
```

Comment:
This one actually has to be with the version, as this is a "final" artifact (similar to the one we add to the release). I think this is the source of the problem behind #359.

Reply:
To be clear: in this PR, this has to be:

```
quay.io/opentelemetry/opentelemetry-operator:v0.31.0
```

The `make bundle VERSION=0.31.0` should generate the right bundle. But doing it might cause the file `config/manager/kustomization.yaml` to be changed, fixing the version on 0.31.0, which is not desired for that file.

Reply:
In time: looks like we never got this bundle right in this repo. I think we always ended up fixing it when sending the PR to the operator hub. Here's what we have for the v0.29.0 tag:

https://github.com/open-telemetry/opentelemetry-operator/blob/a1aacb3b9c64f74650ce4836cfc75bb0da5ad949/bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml#L209

And this for v0.28.0:
https://github.com/open-telemetry/opentelemetry-operator/blob/af398389a1936c0fef3d4e5c94685da71184b9ac/bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml#L209

v0.27.0 looks similar to this PR, but that's still wrong:
https://github.com/open-telemetry/opentelemetry-operator/blob/d4f9c2448b239379a7b2731f0ec1a9c820b10715/bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml#L197